Add tuf-on-ci workflows

jku opened the pull request:
Add workflows for tuf-on-ci (see #1247 and for larger context #929).
* these workflows are copied from root-signing-staging and modified for production environment
* an existing test workflow was renamed because of a naming conflict -- I would rather keep the tuf-on-ci workflow with the original name to make it easier to port future changes
* all scheduled runs are disabled until migration happens
* As an additional safety measure the calls to deploy to GCS are commented out: this way we can enable the workflows without deploying all the way to GCS


These workflows should be complete and ready however:
* they require some additional work before they can be enabled (setting repository variables etc), see #1257 for migration plan
* testing the changes from root-signing-staging can only happen here so there could still be issues: review is very welcome

jku committed:
workflows: Rename test.yml

Same name is used in tuf-on-ci workflows: I would rather rename this one
since the tuf-on-ci workflow is used with workflow_call so
referenced in other workflows

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
Add tuf-on-ci workflows and related custom workflows

These are direct copies from root-signing-staging: the required changes
are done in separate commits to make them visible.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
tuf-on-ci workflows: Update to production

Workflows were copied from staging. Modify them for production:
* Tweak PR and issue templates
* test workflows require URL changes (and some variable name changes)
* test workflows also get a new step where initial root is defined
  (as python-tuf does not consider root < 5 to be valid)
* deploy-to-gcs was modified so the gcloud details are internal
  to the workflow (since in prod it's a separate gcloud project/account,
  not the "tuf-on-ci online signing accout")

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
workflows: Disable tuf-on-ci scheduled runs during migration

* Disable all scheduled runs
* Additionally remove publishing to GCS completely for now

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
workflows: Add sigstore-java to tuf-on-ci custom tests

This test was just added in root-signing-staging so was not in the
initial copy.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

Hayden-IO commented on .github/workflows/ci-test.yml:
```diff
@@ -0,0 +1,110 @@
+#
```
Can you leave a comment or TODO that this will be deleted in the future? (Correct?)

jku replied:
It's correct that this test is likely to be removed after the migration (but that's true for most of the 16 existing workflows)

jku replied:
Actually this one is not entirely obsolete (there's at least yamllint and actionlint in there). I will leave this as is: We'll need to review and clean up the workflows afterwards but that is listed in the migration plan.

jku commented:
last change was a commit message tweak to trigger the broken DCO check

jku commented:
... and I had to rebase because the renamed workflow confused githubs merge resolution

jku committed:
workflows: Embed GCP account details in online-sign

Original design uses GitHub variables: This keeps the online-sign
workflow unchanged from the upstream one and would be nice if the
variables were managed with configuration-as-code. Unfortunately
this is not possible in root-signing.

Embed GCP details in the workflow.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku commented:
I've included the GCP details in the online-sign workflow:

>    Original design uses GitHub variables: This keeps the online-sign
>    workflow unchanged from the upstream one and would be nice if the
>    variables were managed with configuration-as-code. Unfortunately
>    this is not possible in root-signing.
>
>    Embed GCP details in the workflow.

jku commented:
I had to rebase as GitHub uses a conflict resolution that failed in this case (the default one in git rebase just works). No changes.
